src/vehicle.h

```diff
@@ -112,6 +112,8 @@ struct vehicle_part : public JsonSerializer, public JsonDeserializer
 
     vehicle_part( const vpart_str_id& vp, int dx, int dy, item&& it );
 
+    explicit operator bool() const;
```

Comment:
One of the features I never liked: `bool` casting, which does more to obscure the code than to clean it up.
What does it check in this specific case?